core/src/main/java/pl/agh/edu/actor/component/table/BaseTable.java

```diff
@@ -0,0 +1,57 @@
+package pl.agh.edu.actor.component.table;
+
+import java.util.List;
+import java.util.stream.IntStream;
+
+import com.badlogic.gdx.scenes.scene2d.Actor;
+import com.badlogic.gdx.scenes.scene2d.ui.*;
+import com.badlogic.gdx.utils.Align;
+
+import pl.agh.edu.actor.HotelSkin;
+import pl.agh.edu.actor.component.rating.Star;
+import pl.agh.edu.actor.utils.LanguageLabel;
+import pl.agh.edu.actor.utils.WrapperTable;
+
+public abstract class BaseTable extends WrapperTable {
+	private final int noColumns;
+	private final float separatorWidth;
+	private final float cellPadding = 10f;
+	protected final float rowSpacing = 20f;
+
+	public BaseTable(List<String> columnNames) {
+		super();
+
+		this.noColumns = columnNames.size();
+		this.separatorWidth = 2f;
+
+		align(Align.bottomLeft);
+
+		BaseRow header = createRow(columnNames.stream().map(s -> new LanguageLabel(s, "body2")).toArray(Actor[]::new));
+		header.setBackground("table-header-background");
+		header.align(Align.bottomLeft);
+		innerTable.add(header).space(rowSpacing).growX();
+	}
+
+	public final BaseRow createRow(Actor... actors) {
```

Comment:
jak dla mnie ładniejsze było by pojedyncze dodawanie aktorów do BaseRow np dodać mu metodkę add(Actor actor) do klasy i dodawać bo trochę w konstrukorze potworki się robią

Reply:
Gdyby to było złe to James Gosling by inaczej jave stworzył. Nie chcę tworzyć dodatkowej metody na dodawanie pojedyńczych komórek bo dokładnie to już robię wewnątrz BaseRow. Natomiast zamiast varargs mogę przekazywać listę.

Reply:
chodziło mi bardziej o separację BaseTable z BaseRow tak aby np mogliśmy stowrzyć klasę CreditRow z przygotowanym mapperem czyli zamiast 
```java
BaseRow row = createRow(new Label(credit.getTakeOutDate().toString(), HotelSkin.getInstance(), getFont()),
					new Label(credit.getMonthlyPayment().toString(), HotelSkin.getInstance(), getFont()),
					payAllButton);
```

było by w pseduo jezyku coś pokroju

```java
CreditRow extends BaseRow {
        public CreditRow(BankAccount bankAccount){
                  this.add(labelka)
                  this.add(labelka2)
                  this.add(gwiazki)
}
}

przez co CreditTable czyści się z odpowiedzialności pojedynczego wiersza